docs/design/architectural-overview.md

```diff
@@ -0,0 +1,168 @@
+- **[Open Cluster Manager](https://open-cluster-management.io/):** (provides a multi-cluster control plane)
+    - Required.
+- **[Thanos/Prometheus/Grafana](https://thanos.io):** (provide observability integration)
+    - Optional: We look to leverage the existing technology and provide dashboards etc for ingress rather than any specific observability tooling
+- **[Istio.io](https://istio):**
+Gateway API provider that Kuadrant integrates with via WASM and Istio APIS to provide service protection capabilities.
+- Required if you want to use RateLimitPolicy and AuthPolicy
+- **[Gateway API](https://github.com/kubernetes-sigs/gateway-api):**
+New standard for Ingress from the Kubernetes community.
+- required. Core API we integrate with
+
+
+## High Level Multi-Cluster Architecture
+
+![](./images/high-level-multi-cluster.png)
+
+Kuadrant has a multi-cluster gateway controller that is intended to run in a OCM provided "Hub" cluster. This cluster is effectively a central management cluster where policy and gateways along with all that OCM offers can be defined and distributed to the managed "spoke" clusters. 
+
+## Single cluster 
+In a single cluster context, the overall architecture remains the same as above, the key difference is that the Hub and Spoke cluster are now a single cluster rather than multiple clusters. This is how we are initially supporting single cluster.
+
+
+## How does Kuadrant leverage OCM
+
+The Kuadrant multi-cluster gateway controller is intended to be installed into an OCM hub cluster. The Gateway API resources are also installed into this cluster. 
+Kuadrant defines a [gateway class](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/) that the multi-cluster gateway controller is the specified controller for. An OCM user can then (with the correct RBAC in place) define a Gateway just as they would in a single cluster setting and this will be reconciled by the multi-cluster gateway controller. We use this gateway definition to represent a "multi-cluster" enabled gateway and use the aggregated status from each instance of this gateway to store information such as IPs and Hostnames of Load Balancers and how many routes are attached to a given listner.
+
+![](./images/ocm-integration.png)
+
+
+### Gateway Deployment and Distribution
+
+In order for a multi-cluster gateway to be useful, it needs to be distributed or "placed" on a specific set of hub managed spoke clusters.  OCM is responsible for a set of placement and replication APIs. Kuadrant is aware of these APIs, and so when a given gateway is chosen to be placed on a set of managed clusters, Kuadrant multi-cluster gateway controller will ensure the right resources ([ManifestWork](https://open-cluster-management.io/concepts/manifestwork/)) are created in the correct namespaces in the hub. OCM then is responsible for syncing these to the actual spoke cluster and reporting back the status of these resources to the Hub. 
```

Comment:
So what do I have to do as a user to tell I want this gateway in these two cluster?